serializers/shared/src/main/java/io/pravega/schemaregistry/serializer/shared/impl/SerializerFactoryHelper.java

```diff
@@ -42,10 +42,20 @@ private static SchemaRegistryClient getSchemaRegistryClient(SerializerConfig con
             // if auth is enabled and creds are not supplied, reuse the credentials from pravega client config which may
             // be loaded from system properties. 
             SchemaRegistryClientConfig left = config.getRegistryConfigOrClient().getLeft();
-            if (left.isAuthEnabled() && Strings.isNullOrEmpty(left.getAuthMethod())) {
+            if (left.isAuthEnabled() && left.getCredentialProvider() == null) {
                 Credentials creds = ClientConfig.builder().build().getCredentials();
                 left = SchemaRegistryClientConfig.builder().schemaRegistryUri(left.getSchemaRegistryUri())
-                                                 .authentication(creds.getAuthenticationType(), creds.getAuthenticationToken())
+                                                 .credentialProvider(new CredentialProvider() {
```

Comment:
Brilliant.  This will leverage _whatever_ credentials type the Pravega ClientConfig may find in the environment.